src/contact.rs

```diff
@@ -189,6 +190,128 @@ pub async fn make_vcard(context: &Context, contacts: &[ContactId]) -> Result<Str
+            // saved as the authname and not a locally given name.
+            let origin = Origin::OutgoingTo;
+            match Contact::add_or_lookup(context, &contact.authname, &addr, origin).await {
+                Err(e) => return Err(e).context("Contact::add_or_lookup() failed"),
+                Ok((ContactId::SELF, _)) => return Ok(None),
+                Ok(val) => val,
+            }
+        }
+    };
+    if modified != Modifier::None {
+        context.emit_event(EventType::ContactsChanged(Some(id)));
+    }
+    let key = contact.key.as_ref().and_then(|k| {
+        SignedPublicKey::from_base64(k)
+            .with_context(|| {
+                format!(
+                    "import_vcard_contact: Cannot decode key for {}",
+                    contact.addr
+                )
+            })
+            .log_err(context)
+            .ok()
+    });
+    if let Some(public_key) = key {
+        let timestamp = *contact.timestamp.as_ref().unwrap_or(&time());
+        let aheader = Aheader {
+            addr: contact.addr.clone(),
+            public_key,
+            prefer_encrypt: EncryptPreference::NoPreference,
+        };
+        let peerstate = match Peerstate::from_addr(context, &aheader.addr).await {
+            Err(e) => {
+                error!(
+                    context,
+                    "import_vcard_contact: Cannot create peerstate from {}: {e:#}.", contact.addr
+                );
+                return Ok(Some(id));
+            }
+            Ok(p) => p,
+        };
+        let peerstate = if let Some(mut p) = peerstate {
+            p.apply_gossip(&aheader, timestamp);
+            p
+        } else {
+            Peerstate::from_gossip(&aheader, timestamp)
+        };
+        if let Err(e) = peerstate.save_to_db(&context.sql).await {
+            error!(
+                context,
+                "import_vcard_contact: Could not save peerstate for {}: {e:#}.", contact.addr
+            );
+            return Ok(Some(id));
+        }
+        if let Err(e) = peerstate
+            .handle_fingerprint_change(context, timestamp)
+            .await
+        {
+            error!(
+                context,
+                "import_vcard_contact: handle_fingerprint_change() failed for {}: {e:#}.",
+                contact.addr
+            );
+            return Ok(Some(id));
+        }
+    }
+    let path = match &contact.profile_image {
+        Some(image) => match BlobObject::store_from_base64(context, image, "avatar").await {
+            Err(e) => {
+                warn!(
+                    context,
+                    "import_vcard_contact: Could not decode and save avatar for {}: {e:#}.",
+                    contact.addr
+                );
+                None
+            }
+            Ok(path) => Some(path),
+        },
+        None => None,
+    };
+    if let Some(path) = path {
+        let was_encrypted = true; // Currently this doesn't matter.
```

Comment:
It's `true` because we can trust the source, it's not just a random received message, but a vCard imported by user intentionally

Reply:
I tried to improve the comment. As for the test, the last section of `test_make_n_import_vcard()` checks that the self contact isn't imported.

Reply:
> It's `true` because we can trust the source, it's not just a random received message, but a vCard imported by user intentionally

Just because there was an explicit user action, this doesn't mean we can trust the vCard, because it could still have been manipulated. And even very tech-savy users won't manually inspect the vCard before importing it. For now, `import_vcard()` is only for importing Viewtype::Vcard messages - at some later point in time, we can add a feature to import any file and reuse `import_vcard()` for that, but even then, only a very small fraction of users will do this because it requires some technical knowledge, and the vast majority will only ever use this function by clicking on Viewtype::Vcard messages.

I'm not sure how much this matters (and have not a lot of experience with software security), but it seems bad to set `was_encrypted = true` even though we don't know whether it was encrypted, regardless of whether _right now_ this doesn't cause any security problems.

Reply:
I changed it to false, but i'd rather say that the `was_encrypted` param should be renamed so that it says what it actually does e.g. to `allow_selfavatar_chage`, then it's more clear what to pass to the function even if we don't have a message as here